### DISCUSSION TITLE
CanonicalRankAtoms for symmetry returns same rank for different atoms

### DISCUSSION THREAD

**Discussion by julschleinitz** [rdkit/rdkit#7125] [Q&A]
Hi, I am trying to use CanonicalRankAtoms (inspired from discussion: [https://github.com/rdkit/rdkit/discussions/5808](https://github.com/rdkit/rdkit/discussions/5808)) for selecting atoms based on symmetry and I have some issues for a specific class of molecules shown below, the symmetry class does not make chemical sense... would you have suggestions to help with that?

My code:
```
def group_symmetric_atoms(smiles_):
    """
    input : reactant Canonical SMILES
    output:
        mol, Chem.Mol() object annotated with the symmetry group atoms are belonging to
        idx_to_group, dict() with keys being atom idx in the CanonicalSMILES and values beig the label of the group they belong to.
    """
    
    smiles = Chem.CanonSmiles(smiles_)
    mol    = Chem.MolFromSmiles(smiles)
    Chem.RemoveStereochemistry(mol)
    groups = Chem.CanonicalRankAtoms(mol, breakTies=False)
    
    idx_to_group = {}
    
    for at in mol.GetAtoms():
        at.SetProp('atomNote', f"{groups[at.GetIdx()]}")  
        if at.GetSymbol() == 'C':
            idx_to_group.update({at.GetIdx(): groups[at.GetIdx()]})

    return mol, idx_to_group

smiles_list = ['C12C3C4C1C1C2C3C41', 'C12C3C4C1C1C2(O)C3C41', 'C12C3C4C1C1C2N3C41', 'C1CCC(C(C)C)C1']

mols_ = [group_symmetric_atoms(smi)[0] for smi in smiles_list]
Draw.MolsToGridImage(mols_)
```

The ouput is below:

![output](https://github.com/rdkit/rdkit/assets/78845094/436ff5b2-9b5a-4479-a846-7a56ff837e21)

Thanks for your suggestions!

**Comment by greglandrum** [comment]
@truejulosdu13 those results look reasonable to me. What specifically do you think is incorrect?

One possible source of confusion is the drawing of the top molecules. The RDKit's default rendering for cubanes isn't good, so it's easy to get confused if you don't know what you're looking at. The first molecule is a cube:
![image](https://github.com/rdkit/rdkit/assets/540511/1ba60e78-f2ec-45f8-9b15-35f03f197981)
and the others are related structures.

**Reply by julschleinitz** [reply]
Thank you so much, I did not understand it was a cube and overlooked the SMILES, so I thought not all carbons were equivalent and was surprised I only had one class.
This makes sense now.
